wallet/src/de/schildbach/wallet/data/AddressMetadataDao.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022. Dash Core Group.
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.data
+
+import androidx.room.*
+import kotlinx.coroutines.flow.Flow
+import org.bitcoinj.core.Sha256Hash
+import org.dash.wallet.common.transactions.AddressMetadata
+import org.dash.wallet.common.transactions.TaxCategory
+
+/**
+ * @author Eric Britten
+ */
+@Dao
+interface AddressMetadataDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(addressMetadata: AddressMetadata)
+
+    @Query("SELECT * FROM address_metadata")
+    suspend fun load(): List<AddressMetadata>
+
+    @Query("SELECT COUNT(1) FROM address_metadata WHERE address = :address AND isInput = :isInput;")
+    suspend fun exists(address: String, isInput: Boolean): Boolean
+
+    @Query("SELECT * FROM address_metadata WHERE address = :address AND isInput = 0")
+    suspend fun loadRecipient(address: String): AddressMetadata?
+
+    @Query("SELECT * FROM address_metadata WHERE address = :address AND isInput = 1")
+    suspend fun loadSender(address: String): AddressMetadata?
+
+    @Query("INSERT into address_metadata (address, isInput, taxCategory) VALUES (:address, :isInput, :taxCategory)")
+    suspend fun markAddress(address: String, isInput: Boolean, taxCategory: TaxCategory)
```

Comment:
this duplicates the `insert` function above and probably fails if there is already an entry.  Theoretically, the same address wouldn't be used twice.

Reply:
Some of these functions may not be used and could be removed when before the PR is closed out.